Sources/Client/Client/Client+Events.swift

```diff
@@ -0,0 +1,150 @@
+    /// Cancel the underlying subscription for this object.
+    func cancel()
+}
+
+/// Reference for the subscription initiated. Call `cancel()` to end the subscription. Alternatively, when this object is deallocated, it calls `cancel()` on itself automatically.
+public protocol AutoCancellable: Cancellable {}
+
+struct Subscription: Cancellable {
+    private let onCancel: (String) -> ()
+    let uuid: String
+    
+    init(onCancel: @escaping (String) -> ()) {
+        self.onCancel = onCancel
+        uuid = UUID().uuidString
+    }
+    
+    public func cancel() {
+        onCancel(uuid)
+    }
+}
+
+class SubscriptionBag: Cancellable {
+    private var subscriptions = [Cancellable]()
+    
+    public func add(_ subscription: Cancellable)  {
+        subscriptions.append(subscription)
+    }
+    
+    public func cancel() {
+        subscriptions.forEach { $0.cancel() }
+    }
+}
+
+extension Client {
+    /// Observe events for the given event types.
+    /// - Parameters:
+    ///   - eventTypes: A set of event types to be observed. Defaults to all events.
+    ///   - callback: Callback closure to be called for each new event.
+    /// - Returns: `Subscription` object to be able to cancel observing. Call `subscription.cancel()` when you want to stop observing.
+    /// - Warning: Subscriptions do not cancel on `deinit` and that can cause crashes / memory leaks, so make sure you handle subscriptions correctly.
+    public func subscribe(forEvents eventTypes: Set<EventType> = Set(EventType.allCases), _ callback: @escaping OnEvent) -> Cancellable {
+        subscribe(forEvents: eventTypes, cid: nil, callback)
+    }
+    
+    func subscribe(forEvents eventTypes: Set<EventType> = Set(EventType.allCases), cid: ChannelId?, _ callback: @escaping OnEvent) -> Cancellable {
+        let subscription = Subscription { [unowned self] uuid in
+            self.eventHandlingQueue.async {
+                self.onEventObservers[uuid] = nil
+            }
+        }
+        
+        let handler: OnEvent = { event in
+            guard eventTypes.contains(event.type) else {
+                return
+            }
+            
+            if let cid = cid, event.cid != cid {
+                return
+            }
+            
+            callback(event)
+        }
+        
+        onEventObservers[subscription.uuid] = handler
+        
+        return subscription
+    }
+    
+    public func subscribeToUserUpdates(_ callback: @escaping OnUpdate<User>) -> Cancellable {
+        let subscription = Subscription { [unowned self] uuid in
+            self.eventHandlingQueue.async {
+                self.onUserUpdateObservers[uuid] = nil
+            }
+        }
+        
+        onUserUpdateObservers[subscription.uuid] = callback
+        
+        return subscription
+    }
+    
+    public func subscribeToUnreadCount(_ callback: @escaping OnUpdate<UnreadCount>) -> Cancellable {
+        subscribeToUserUpdates { (user) in
```

Comment:
Please, can you omit rounded brackets, since we don't do that for `if` as well, it's optional. They only needed for var name with type.